uds/uds_config_tool/FunctionCreation/ReadDataByIdentifierMethodFactory.py

```diff
@@ -100,182 +84,75 @@ def create_requestFunctions(diagServiceElement, xmlElements):
+
                     elif dataObjectElement.tag == "STRUCTURE":
-                        start = int(param.find("BYTE-POSITION").text)
-                        listLength = int(dataObjectElement.find("BYTE-SIZE").text)
-                        totalLength += listLength
+                        diagCodedType = getDiagCodedTypeFromStructure(dataObjectElement, xmlElements)
                     else:
+                        # neither DOP nor STRUCTURE
                         pass
+                    param = Param(shortName, bytePosition, diagCodedType)
+                    params.append(param)
                 else:
+                    # not a PARAM with SID, ID (= DID), or DATA
                     pass
-            except:
-                # print(sys.exc_info())
-                pass
-
-        checkSIDRespFuncString = checkSIDRespFuncTemplate.format(
-            checkSIDRespFuncName,  # 0
-            responseId,  # 1
-            responseIdStart,  # 2
-            responseIdEnd,
-        )  # 3
-        exec(checkSIDRespFuncString)
-        checkSIDLenFuncString = checkSIDLenFuncTemplate.format(
-            checkSIDLenFuncName, SIDLength  # 0
-        )  # 1
-        exec(checkSIDLenFuncString)
-        checkDIDRespFuncString = checkDIDRespFuncTemplate.format(
-            checkDIDRespFuncName,  # 0
-            diagnosticId,  # 1
-            diagnosticIdStart
-            - SIDLength,  # 2... note: we no longer look at absolute pos in the response,
-            diagnosticIdEnd - SIDLength,
-        )  # 3      but look at the DID response as an isolated extracted element.
-        exec(checkDIDRespFuncString)
-        checkDIDLenFuncString = checkDIDLenFuncTemplate.format(
-            checkDIDLenFuncName, totalLength - SIDLength  # 0
-        )  # 1
-        exec(checkDIDLenFuncString)
-
-        return (
-            locals()[checkSIDRespFuncName],
-            locals()[checkSIDLenFuncName],
-            locals()[checkDIDRespFuncName],
-            locals()[checkDIDLenFuncName],
-        )
-
-    ##
-    # @brief may need refactoring to deal with multiple positive-responses (WIP)
-    @staticmethod
-    def create_encodePositiveResponseFunction(diagServiceElement, xmlElements):
-
-        positiveResponseElement = xmlElements[
-            (diagServiceElement.find("POS-RESPONSE-REFS"))
-            .find("POS-RESPONSE-REF")
-            .attrib["ID-REF"]
-        ]
-
-        shortName = diagServiceElement.find("SHORT-NAME").text
-        encodePositiveResponseFunctionName = "encode_{0}".format(shortName)
-
-        params = positiveResponseElement.find("PARAMS")
-
-        encodeFunctions = []
-
-        for param in params:
-            try:
-                semantic = None
-                try:
-                    semantic = param.attrib["SEMANTIC"]
-                except AttributeError:
-                    pass
+            except Exception as e:
+                log.debug(e)
```

Comment:
Should this really be debug information?

Reply:
Changed it to traceback.print_tb(). Originally, it did not give any hint an exception was thrown, which was hard to debug. So I don't want to just `pass` it as it was before.

Reply:
removed the try except in create_positive_response_objects, changed logs to a warning in create_checkNegativeResponseFunctions